tests/manager/repositories/scaling_group/test_scaling_group_repository.py

```diff
@@ -0,0 +1,331 @@
+        scaling_group_names = []
+        async with db_with_cleanup.begin_session() as db_sess:
+            for i in range(20):
+                sgroup_name = f"test-sgroup-mixed-{i:02d}"
+                sgroup = ScalingGroupRow(
+                    name=sgroup_name,
+                    description=f"Test scaling group {i:02d}",
+                    is_active=(i % 2 == 0),  # Even indexes active
+                    is_public=True,
+                    created_at=datetime.now(),
+                    wsproxy_addr=None,
+                    wsproxy_api_token=None,
+                    driver="static",
+                    driver_opts={},
+                    scheduler="fifo",
+                    scheduler_opts=ScalingGroupOpts(),
+                    use_host_network=False,
+                )
+                db_sess.add(sgroup)
+                scaling_group_names.append(sgroup_name)
+            await db_sess.flush()
+
+        yield scaling_group_names
+
+    @pytest.fixture
+    async def sample_scaling_groups_medium(
+        self,
+        db_with_cleanup: ExtendedAsyncSAEngine,
+    ) -> AsyncGenerator[list[str], None]:
+        """Create 15 sample scaling groups for no-pagination testing"""
+        scaling_group_names = []
+        async with db_with_cleanup.begin_session() as db_sess:
+            for i in range(15):
+                sgroup_name = f"test-sgroup-medium-{i}"
+                sgroup = ScalingGroupRow(
+                    name=sgroup_name,
+                    description=f"Test scaling group {i}",
+                    is_active=True,
+                    is_public=True,
+                    created_at=datetime.now(),
+                    wsproxy_addr=None,
+                    wsproxy_api_token=None,
+                    driver="static",
+                    driver_opts={},
+                    scheduler="fifo",
+                    scheduler_opts=ScalingGroupOpts(),
+                    use_host_network=False,
+                )
+                db_sess.add(sgroup)
+                scaling_group_names.append(sgroup_name)
+            await db_sess.flush()
+
+        yield scaling_group_names
+
+    @pytest.fixture
+    async def scaling_group_repository(
+        self,
+        db_with_cleanup: ExtendedAsyncSAEngine,
+    ) -> AsyncGenerator[ScalingGroupRepository, None]:
+        """Create ScalingGroupRepository instance with database"""
+        repo = ScalingGroupRepository(db=db_with_cleanup)
+        yield repo
+
+    @pytest.mark.asyncio
+    async def test_search_scaling_groups_all(
+        self,
+        scaling_group_repository: ScalingGroupRepository,
+        sample_scaling_groups_small: list[str],
+    ) -> None:
+        """Test searching all scaling groups without filters"""
+        result = await scaling_group_repository.search_scaling_groups()
+
+        # Should have at least the 5 test scaling groups
+        assert len(result.items) >= 5
+        assert result.total_count >= 5
```

Comment:
Now that we create fixtures every time, we can test based on the exact number, whereas previously, when we had a common fixture, we checked `ge` because we didn't test strictly.